src/methods/validate-fixture-input.ts

```diff
@@ -65,22 +82,29 @@ export function validateFixtureInput(
           let possibleTypes = new Set(
             possibleTypesStack[possibleTypesStack.length - 1],
           );
+          let namedType: GraphQLNamedType | undefined;
+
           if (node.typeCondition !== null && node.typeCondition !== undefined) {
-            const namedType = schema.getType(node.typeCondition.name.value);
-            if (namedType && isAbstractType(namedType)) {
-              possibleTypes = possibleTypes.intersection(
-                new Set(
-                  schema.getPossibleTypes(namedType).map((type) => type.name),
-                ),
-              );
-            } else if (namedType && isObjectType(namedType)) {
-              possibleTypes = new Set([namedType.name]);
+            namedType = schema.getType(node.typeCondition.name.value);
+
+            if (namedType) {
+              if (isAbstractType(namedType)) {
+                possibleTypes = possibleTypes.intersection(
+                  new Set(
+                    schema.getPossibleTypes(namedType).map((type) => type.name),
+                  ),
+                );
+              } else if (isObjectType(namedType)) {
+                possibleTypes = new Set([namedType.name]);
+              }
             }
           }
           possibleTypesStack.push(possibleTypes);
+          typeConditionStack.push(namedType ?? null);
```

Comment:
This is a bit of an `undefined` vs `null` thing.

schema.getType can return undefined, but in selection sets I explicitly push `null` values to the typeConditionStack. I chose `null` to represent that I was consciously setting the value to "nothing", and we branch through the expectedFields setting logic with that.

... I could have just used `undefined`, but I felt `null` was a clearer representation of what I was doing.

This way we keep the graphql concept of "i tried to look up a type and it wasn't defined" and the stack concept of "am I in an inlinefragment" logic separate.